src/reducers/favorites-reducer.js

```diff
@@ -0,0 +1,21 @@
+import {ADDED_FAVORITE, DELETED_FAVORITE} from '../actions/types'
+
+const initialFavorites = JSON.parse(localStorage.getItem('favorites')) || {}
+
+export default function(state=initialFavorites, action) {
+  let newState
+  switch(action.type){
+    case ADDED_FAVORITE:
+      newState = {...state, [action.payload.id]: action.payload}
+      localStorage.setItem('favorites', JSON.stringify(newState))
```

Comment:
Remember that reducers must be pure, the calls to `localStorage` are side effects.